### ISSUE TITLE
Regarding tuning the prefixes

### ISSUE THREAD

**Issue by base-y** [HKUNLP/UnifiedSKG#32]
@Timothyxxx  Hi, I have few queries regarding prefix tuning. Can you shed your opinions on the same since you have worked extensively on it (the github repo of Prefix tuning seems abandoned).

1. Will training the whole model along with prefixes yield a better performance (assuming its not low data regime - greater than 1k datapoints)?
2. Assume we train the prefixes in two cases: freezing the model and not freezing the model. Will the learned representation of prefixes for both the cases be the same or different?
3. If I train the model+prefixes in two scenarios: (i) train the model and prefixes together (finetune - single shot) and, (ii) finetune the model only first, freeze it, and then train the prefixes only with the frozen weights (two stages). Is there any difference between both of them (I think there is) and would like to know about why the diff?

**Comment by Timothyxxx** [comment]
Hi, thanks for asking!

1. It is generally tested, that prompt methods is better when we only have few training data. But we have no clue what is the number of examples when it is enough also for finetuning.
2. Of course not the same.
3. To be honest, we tried all your thoughts before, and our results showed(at least in SKG tasks),  the setting i and setting ii got very similar performance. You can try yourself again to test if we were missing anything.

Hope these information helpful!

Thanks!

**Comment by base-y** [comment]
@Timothyxxx Thank you for the answers. But for answer 1, I understand the model performance will vary because the LM model is updated when the loss is propagated backward. But I think the same gradient will flow to the prefix representation (prefix learnable parameter) in both the cases. In this case, though the model performance might be different, the final prefix representations will be the same I guess (since they are updated with the same gradient at every step irrespective of whether the model is frozen or not)?

**Comment by Timothyxxx** [comment]
I still don't understand, I recommend try yourself and compare the weight of two LM (frozen/un-frozen).
